crates/hyperswitch_connectors/src/connectors/nexixpay.rs

```diff
@@ -115,43 +125,96 @@ impl ConnectorCommon for Nexixpay {
     ) -> CustomResult<Vec<(String, masking::Maskable<String>)>, errors::ConnectorError> {
         let auth = nexixpay::NexixpayAuthType::try_from(auth_type)
             .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
-        Ok(vec![(
-            headers::AUTHORIZATION.to_string(),
-            auth.api_key.expose().into_masked(),
-        )])
+        Ok(vec![
+            (
+                headers::X_API_KEY.to_string(),
+                auth.api_key.expose().into_masked(),
+            ),
+            (
+                headers::CORRELATION_ID.to_string(),
+                Uuid::new_v4().to_string().into_masked(),
```

Comment:
Shouldn't this ideally be something that we can debug with and we also have at any point in time, like our request ID or our attempt ID? If we generate a new UUID and share it with the connector, but we don't store it, won't this be lost?

Reply:
as discussed, We will be keeping correlationId as random uuid as we don't want to use x-request-id to be stored and used for debugging as this id cannot be used for further queries. Also request Id will be changing in subsequent calls which will not serve the purpose here of correlationId.